src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -229,30 +329,56 @@ public function get_product_by_metas( $metas = array() ) {
 		$where   = array();
 		$results = array();
 		$params  = array();
-
 		foreach ( $metas as $column => $value ) {
+			if ( empty( $value ) ) {
+				continue;
+			}
+
+			$value = is_array( $value ) ? implode( ',', $value ) : $value;
+
+			if ( 'stock_status' === $column ) {
+				$stock_product_ids = $wpdb->get_col(
+					$wpdb->prepare(
+						"SELECT DISTINCT product_id FROM {$wpdb->prefix}wc_product_meta_lookup WHERE stock_status = %s",
+						$value
+					)
+				);
+				continue;
+			}
+
 			if ( 'min_price' === $column ) {
-				$where[]  = "{$column} >= %f";
-				$params[] = (float) $value;
+				$where[]  = "{$column} >= %d";
+				$params[] = intval( $value ) / 100;
 				continue;
 			}
 
 			if ( 'max_price' === $column ) {
-				$where[]  = "{$column} <= %f";
-				$params[] = (float) $value;
+				$where[]  = "{$column} <= %d";
+				$params[] = intval( $value ) / 100;
 				continue;
 			}
 
-			$where[]  = "{$column} = %s";
+			if ( 'average_rating' === $column ) {
+				$value   = is_array( $value ) ? implode( ',', $value ) : $value;
+				$where[] = sprintf( 'average_rating BETWEEN %s - 0.5 AND %s + 0.5', $value, $value );
```

Comment:
I guess I am still not 100% sure the half up/down is needed? `BETWEEN` is inclusive already.

Reply:
> I guess I am still not 100% sure the half up/down is needed? BETWEEN is inclusive already.

Yes, it's true that BETWEEN is inclusive, which is a problem; as an example: if the average rating is precisely 2.5, with `BETWEEN`, both the ratings 2 and 3 would be counted, but actually, we want just the 3 to be counted because that's the default behavior of the `round()` function. 

As a very basic example:

<img width="657" alt="Screenshot 2023-04-04 at 18 01 54" src="https://user-images.githubusercontent.com/15730971/229921225-d2cca051-b6a4-4531-aa55-3d01386249d3.png">

Reply:
Interesting. When I tested, if I use `2.5`, I don't get `2` also.

![iutlW3.png](https://user-images.githubusercontent.com/2132595/230085616-ce682b86-3d67-4e3e-bc0e-55a72c529c2e.png)

Reply:
> Interesting. When I tested, if I use 2.5, I don't get 2 also.

Hm, I'm not sure if I follow: out of curiosity, how would you re-write this query with a `BETWEEN` and also circumvent the problem of having the same product counted multiple times for different intervals?

Using your query as an example on a shop with a product with average rating = 2.5 and another one with average rating = 3.5:

```sql
select distinct product_id, average_rating from wp_wc_product_meta_lookup where average_rating BETWEEN 1.5 and 2.5;
select distinct product_id, average_rating from wp_wc_product_meta_lookup where average_rating BETWEEN 2.5 and 3.5;
select distinct product_id, average_rating from wp_wc_product_meta_lookup where average_rating BETWEEN 3.5 and 4.5;
```

Results for the first query:
<img width="1234" alt="Screenshot 2023-04-05 at 16 24 27" src="https://user-images.githubusercontent.com/15730971/230185915-d51a2fa5-836f-4331-9e4e-988a1b4045d1.png">

Results for the second query:
<img width="1235" alt="Screenshot 2023-04-05 at 16 24 58" src="https://user-images.githubusercontent.com/15730971/230185954-ddd133d5-7ee3-4a27-8178-303c67e9919c.png">

Results for the third query:
<img width="1234" alt="Screenshot 2023-04-05 at 16 26 12" src="https://user-images.githubusercontent.com/15730971/230185997-23515716-d606-4750-814f-4aaaf46638a1.png">


In this scenario, product ID 15 comes up on the count results for the first and the second queries. Also, product ID 12 comes up on the count results for the second and third queries, when they shouldn't, as this way the counts are duplicated/incorrect.

Reply:
Perhaps we could `array_unique` on the result on every loop so it will not contain any duplicates. Just a thought as an option.

Reply:
> Perhaps we could array_unique on the result on every loop so it will not contain any duplicates. Just a thought as an option.

Hmm, but that would not solve the problem either. How would you define the correct count for each one of the results via PHP? In our example here, how to determine if Product ID 15 belongs to the first or the second queries after they were already fetched from the DB? Also, why would we prefer to fetch incorrect results from the DB to work on their logic later via PHP (assuming there's a way to do so reliably)?

The counts are working as expected the way they are structured now: I'm glad to adopt alternative solutions if they are better in terms of efficiency/performance, but I'm not seeing the usage of `BETWEEN` as a way to achieve this yet. How about pushing these changes to a separate branch to compare both approaches?